Snail/Disposer.swift

```diff
@@ -0,0 +1,16 @@
+//  Copyright © 2019 Compass. All rights reserved.
+
+import Foundation
+
+public protocol DisposableType {
+    func dispose()
+}
+
+public class Disposer {
+    public var disposables: [DisposableType] = []
```

Comment:
Does it make sense to have this be `private(set)` and add a method like `add(disposable: DisposableType)`

Reply:
it totally does. Pushed!